src/raftstore/store/peer.rs

```diff
@@ -1050,6 +1080,24 @@ impl Peer {
         }
     }
 
+    pub fn apply_prepare_merge(&mut self, region: metapb::Region, state: MergeState) {
+        // Set pending_merge to true, in case it reboots and a committed
+        // prepare merge is not applied yet.
+        self.pending_merge_idx = Some(state.get_commit());
+
+        self.pending_merge_state = Some(state);
+        self.set_region(region);
+    }
+
+    pub fn apply_rollback_merge(&mut self, region: Option<metapb::Region>) {
```

Comment:
So when `region = None` it is intended that the region is not set?

Reply:
It means no need to update `region`.